rpk: add labels to rpk generate prometheus-config

r-vasquez opened the pull request:
This PR introduces a new flag '--labels' which
lets the user add a common label for the targets
in the rpk-generated scrape-config.

Fixes #5548

### Example:

Common tag
```yaml
$ rpk generate prometheus-config --job-name redpanda-metrics-test --node-addrs 'localhost:9644,localhost:9645' --internal-metrics --labels "hello=world" 
- job_name: redpanda-metrics-test
  static_configs:
    - targets:
        - localhost:9644
        - localhost:9645
      labels:
        hello: world
  metrics_path: /public_metrics
- job_name: redpanda-metrics-test
  static_configs:
    - targets:
        - localhost:9644
        - localhost:9645
      labels:
        hello: world
  metrics_path: /metrics
```
Different tags per job
```yaml
$ rpk generate prometheus-config --job-name redpanda-metrics-test --node-addrs 'localhost:9644,localhost:9645' --internal-metrics --labels "internal:foo=bar,public:hello=world"
- job_name: redpanda-metrics-test
  static_configs:
    - targets:
        - localhost:9644
        - localhost:9645
      labels:
        hello: world
  metrics_path: /public_metrics
- job_name: redpanda-metrics-test
  static_configs:
    - targets:
        - localhost:9644
        - localhost:9645
      labels:
        foo: bar
  metrics_path: /metrics
```

## Backports Required

<!-- Checking at least one of the checkboxes is REQUIRED if this PR is not a backport. -->

- [x] none - not a bug fix

## Release Notes
### Improvements

* now you can specify labels in the rpk-generated Prometheus config via the `labels` flag.

gene-redpanda reviewed (dismissed):
LGTM

r-vasquez commented:
Unrelated Failures

|Build|Issue|
|--|--|
|https://buildkite.com/redpanda/redpanda/builds/31317#0188bb9c-7f22-4341-bb88-d60f34c48c0f/6-6631|https://github.com/redpanda-data/redpanda/issues/11306|
|https://buildkite.com/redpanda/redpanda/builds/31317#0188bb7e-46df-41db-a768-aebe518f818b/363-5766|https://github.com/redpanda-data/redpanda/issues/11454|

twmb commented on src/go/rpk/pkg/cli/generate/prometheus.go:
```diff
@@ -234,3 +272,43 @@ func createClient(fs afero.Fs, addr string, tlsCfg tlsConfig) (*kgo.Client, erro
+	ext = make(map[string]string)
+
+	for _, label := range labels {
+		if s := strings.Split(label, "|"); len(s) > 1 {
```
SplitN(label, "|", 2)

twmb commented on src/go/rpk/pkg/cli/generate/prometheus.go:
```diff
@@ -42,60 +47,91 @@ type prometheusTLS struct {
+section using the --labels flag. You can also specify which metric you want the 
+label to target. For example:
+
+  --job-name test --labels "ext|group:one,int|group:two"
```
This help text doesn't say what ext or int are, and I don't understand how they map to the yaml in the comment just below?

r-vasquez replied:
Changed, moved the help text to a variable.

r-vasquez commented:
[Force Push](https://github.com/redpanda-data/redpanda/compare/393c6f0a5897e68dc1f3d16a158d2a342f569e6e..62057b2cdc4cdf18f6348c50b1d5cca92164db3c) changes:
- Moved the help text to a constant and explained what every 'metric targets' means.
- Changed metric target from `ext` to `pub` which makes more sense since it targets public_metrics.

r-vasquez commented:
[Force push](https://github.com/redpanda-data/redpanda/compare/62057b2cdc4cdf18f6348c50b1d5cca92164db3c..62783520e2fe6511c4bd91f2d28dde0659627495) changes:
- Changed the format to be `metric:labelName=labelValue`

r-vasquez commented:
**Unrelated Failures**
|Build|Issue|
|--|--|
|https://buildkite.com/redpanda/redpanda/builds/31484#0188c525-8ae6-409e-97ba-879eb5d121f8/6-6050|https://github.com/redpanda-data/redpanda/issues/11304|
|https://buildkite.com/redpanda/redpanda/builds/31484#0188c508-58d1-415f-948b-85f270d1a808/6-5465|https://github.com/redpanda-data/redpanda/issues/11276|
